sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/logicaltypes/FixedBytes.java

```diff
@@ -43,44 +42,15 @@ public int getLength() {
     return byteArraySize;
   }
 
-  @Override
-  public String getIdentifier() {
-    return IDENTIFIER;
-  }
-
-  @Override
-  public FieldType getArgumentType() {
-    return FieldType.INT32;
-  }
-
-  @Override
-  public Integer getArgument() {
-    return byteArraySize;
-  }
-
-  @Override
-  public FieldType getBaseType() {
-    return FieldType.BYTES;
-  }
-
   @Override
   public byte[] toBaseType(byte[] input) {
-    checkArgument(input.length == byteArraySize);
+    checkArgument(input == null || input.length == byteArraySize);
     return input;
   }
 
   @Override
   public byte[] toInputType(byte[] base) {
-    checkArgument(base.length <= byteArraySize);
-    if (base.length == byteArraySize) {
-      return base;
-    } else {
-      return Arrays.copyOf(base, byteArraySize);
-    }
-  }
```

Comment:
FYI @reuvenlax this PR is removing the ability to convert smaller byte arrays. As noted in https://github.com/apache/beam/pull/11609#issuecomment-624810068 it seems this logic is inaccessible anyway.